lib/housing-wanted/housing-wanted.controllers.js

```diff
@@ -1,7 +1,118 @@
 module.exports = (models) => {
     const Boom = require('boom');
+    const Sequelize = require('sequelize');
+    const Op = Sequelize.Op;
+    const housingWantedModels = require('./housing-wanted.models');
+    const accountLib = require('../account/account.controllers')(models).lib;
+    const locationLib = require('../location/location.controllers')(models).lib;
+    const Log = require('../log/log.controllers')(models).lib;
   
-    return {}
+    return {
+      postHousingWanted: async function(request, h){
+        /*
+          # Require Account for submitting fields other than address
+        */
+        let payload = request.payload;
+        let account;
+        try {
+         account = await accountLib.getAccount(request.auth.credentials);
+        } catch (e) {
+         throw Boom.badImplementation('Error during getAccount(request.auth.credentials). ' + e);
+       }
+
+        if (account === null) {
+          throw Boom.badRequest('Must have an Account to create a housing wanted post');
+        }
+
+        /* Restrict AuthorId to same user, or SuperAdmin */
+        if (payload.AuthorId) {
+          // If AuthorId is set then we have already passed auth and account lookup
+          // This is because if AuthorId is set then not only quickInfo is set
+          if (
+            payload.AuthorId !== account.id  &&
+            request.auth.credentials.subjectId !== process.env.SUPER_ADMIN
+          ) {
+            throw Boom.badRequest('HousingWanted.AuthorId must match your Account ID or you must be a Super Admin');
+          }
+        }
+
+        // bathrooms validation
+        let hasBathrooms = false;
+        let bathrooms = Number(request.payload.bathrooms);
+        if (bathrooms) {
+          hasBathrooms = true;
+        }
+
+        // number of bathrooms should be set
+        // return true if bathrooms object has specified property
+        if (request.payload.hasOwnProperty("bathrooms")){
+          hasBathrooms = true;
+        }
+
+        // bathrooms can be in .25 increments
+        if (hasBathrooms && (!bathroomsValidIncrement(bathrooms)) ) {
+          throw Boom.badData('bathrooms must be in .25 increments');
+        }
+
+        // bedrooms validation
+        let hasBedrooms = false;
+        let bedrooms = Number(request.payload.bedrooms);
+        // return true if bedrooms object has specified property
+        if (request.payload.hasOwnProperty("bedrooms")){
+          hasBedrooms = true;
+        }
+
+        // start building housingWanted object
+        let housingWantedObject = {
+          AuthorId: account.id,
+          title: request.payload.title,
+          bedrooms: bedrooms,
+          bathrooms: bathrooms,
+          contact: request.payload.contact,
+          body: request.payload.body,
+          status: request.payload.status,
+          details: request.payload.details,
+        };
```

Comment:
Building the object this way is really only suitable when all the fields are required/guaranteed to be there, because it makes the assumption they are. For example, this assigns bedrooms and bathrooms, but they aren't required, which results in them being Number(undefined) which = NaN, which will fail on db insert. It also sets AuthorId to account.id regardless of whether AuthorId was submitted or not. 

The preferred way of doing this would be to create the housingWantedObject, and assign each property in a conditional that checks the appropriate thing. eg:
if (hasBathrooms) { housingWantedObject.bathrooms = bathrooms; }
if (request.AuthorId) { housingWantedObject.AuthorId = request.AuthorId; } else { housingWantedObject.AuthorId = account.id; }